client/src/views/JoinOrganizationModal.vue

```diff
@@ -0,0 +1,495 @@
+  <!-- Components: 1 part = 1 component -->
+
+  <ion-page class="modal">
+    <ion-buttons
+      slot="end"
+      class="closeBtn-container"
+    >
+      <ion-button
+        slot="icon-only"
+        @click="cancelModal()"
+        class="closeBtn"
+        v-show="canClose()"
+      >
+        <ion-icon
+          :icon="close"
+          size="large"
+          class="closeBtn__icon"
+        />
+      </ion-button>
+    </ion-buttons>
+    <ion-header class="modal-header">
+      <wizard-stepper
+        v-if="pageStep > JoinOrganizationStep.WaitForHost"
+        :current-index="pageStep - 1"
+        :titles="[
+          $t('JoinOrganization.stepTitles.GetHostCode'),
+          $t('JoinOrganization.stepTitles.ProvideGuestCode'),
+          $t('JoinOrganization.stepTitles.ContactDetails'),
+          $t('JoinOrganization.stepTitles.Password'),
+          $t('JoinOrganization.stepTitles.Validation')
+        ]"
+      />
+
+      <ion-title
+        v-if="titles.get(pageStep)?.title !== ''"
+        class="modal-header__title title-h2"
+      >
+        <ion-icon
+          :icon="informationCircle"
+        />
+        {{ titles.get(pageStep)?.title }}
+      </ion-title>
+      <ion-text
+        v-if="titles.get(pageStep)?.subtitle !== ''"
+        class="modal-header__text body"
+      >
+        {{ titles.get(pageStep)?.subtitle }}
+      </ion-text>
+    </ion-header>
+    <!-- modal content: create component for each part-->
+    <div class="modal-content inner-content">
+      <!-- part 1 (wait for host)-->
+      <div
+        v-show="pageStep === JoinOrganizationStep.WaitForHost"
+        class="step orga-name"
+      >
+        <wait-for-host-page
+          ref="waitPage"
+          :org-name="'My Org'"
+        />
+      </div>
+
+      <!-- part 2 (host code)-->
+      <div
+        v-show="pageStep === JoinOrganizationStep.GetHostSasCode"
+        class="step"
+      >
+        <sas-code-choice
+          ref="hostCodePage"
+          :choices="['ABCD', 'EFGH', 'IJKL', 'MNOP']"
+          @select="selectHostSas($event)"
+        />
+      </div>
+
+      <!-- part 3 (guest code)-->
+      <div
+        v-show="pageStep === JoinOrganizationStep.ProvideGuestCode"
+        class="step"
+      >
+        <sas-code-provide
+          ref="guestCodePage"
+          :code="'ABCD'"
+        />
+      </div>
+
+      <!-- part 4 (user info)-->
+      <div
+        v-show="pageStep === JoinOrganizationStep.GetUserInfo"
+        class="step"
+        id="get-user-info"
+      >
+        <user-information-page
+          ref="userInfoPage"
+          :email-enabled="false"
+          :device-enabled="!waitingForHost"
+          :name-enabled="!waitingForHost"
+        />
+      </div>
+      <!-- part 5 (get password)-->
+      <div
+        v-show="pageStep === JoinOrganizationStep.GetPassword"
+        class="step"
+        id="get-password"
+      >
+        <choose-password
+          ref="passwordPage"
+        />
+      </div>
```

Comment:
The page becomes very slow when you paste a long password like this: parsec://parsec.cloud/Test?action=claim_user&token=47265123969c4d6584c2bc15960cf212
After a while, the situation returns to normal, but if you try to add characters at the end, they are not displayed for a few seconds. 

https://github.com/Scille/parsec-cloud/assets/81075511/5895207f-1844-4a05-89d9-dd4bad9dbe22

Reply:
Caused by zxcvbn with English and French dictionnaries called multiple times every time the password changed. I removed the dictionnaries so that it would be faster, and added a variable to cache the password strength so it wouldn't be called multiple times.